### PR TITLE
[14.0] edi_oca: misc fix/imp

### PR DESCRIPTION
**Pull request by simahawk**
See atomic commits for details.

### CODE REVIEW

File: edi_oca/models/edi_backend.py [hunk 1]
```diff
@@ -341,7 +341,7 @@ def _cron_check_output_exchange_sync(self, **kw):
             backend._check_output_exchange_sync(**kw)
 
     # TODO: consider splitting cron in 2 (1 for receiving, 1 for processing)
-    def _check_output_exchange_sync(self, skip_send=False):
+    def _check_output_exchange_sync(self, skip_send=False, skip_sent=True):
```
Comment (TDu) [inline edi_oca/models/edi_backend.py:344]: The docstring could be updated for the new parameter

File: edi_exchange_template_oca/models/edi_exchange_template_mixin.py [hunk 2]
```diff
@@ -13,6 +13,19 @@
 _logger = logging.getLogger(__name__)
 
 
+def date_to_datetime(dt):
+    """Convert date to datetime."""
+    if isinstance(dt, datetime.date):
+        return datetime.datetime.combine(dt, datetime.datetime.min.time())
+    return dt
```
Comment (SilvioC2C) [inline edi_exchange_template_oca/models/edi_exchange_template_mixin.py:20]: Could it make sense to add timezone info?
Also, `datetime.datetime.min.time()` can be replaced by importing `time` from `datetime` and then using `time.min`.

File: edi_exchange_template_oca/models/edi_exchange_template_mixin.py [hunk 3]
```diff
@@ -77,22 +90,42 @@ def _utc_now():
...
     def _date_to_string(dt, utc=True):
+        if not dt:
+            return ""
         if utc:
-            dt = dt.astimezone(pytz.UTC)
+            dt = to_utc(dt)
         return fields.Date.to_string(dt)
 
+    @staticmethod
+    def _datetime_to_string(dt, utc=True):
```
Comment (SilvioC2C) [inline edi_exchange_template_oca/models/edi_exchange_template_mixin.py:100]: `_date_to_string` and `_datetime_to_string` are the same method, maybe we can avoid defining one of them and simply execute the other one when called. Something like this:
```
    @staticmethod
    def _date_to_string(dt, utc=True):
        if not dt:
            return ""
        if utc:
            dt = to_utc(dt)
        return fields.Date.to_string(dt)

    _datetime_to_string = _date_to_string
```

File: edi_oca/models/edi_exchange_consumer_mixin.py [hunk 4]
```diff
@@ -232,3 +238,10 @@ def get_edi_access(self, doc_ids, operation, model_name=False):
         else:
             check_operation = operation
         return check_operation
+
+    def _edi_set_origin(self, exc_record):
+        self.sudo().update({"origin_exchange_record_id": exc_record.id})
+
+    def _edi_get_origin(self):
+        self.ensure_one()
+        return self.origin_exchange_record_id
```
Comment (SilvioC2C) [inline edi_oca/models/edi_exchange_consumer_mixin.py:247]: Any specific reason for having setter/getter methods for this field?

**Review by SilvioC2C — APPROVED** [review]
Overall LGTM, just some minor comments

File: edi_oca/models/edi_exchange_consumer_mixin.py (hunk 4 above)
Comment (simahawk) [reply edi_oca/models/edi_exchange_consumer_mixin.py:247]: they are not setter/getter for a field, they are api methods. And no that you let me think about them, they should not be private.

File: edi_exchange_template_oca/models/edi_exchange_template_mixin.py (hunk 3 above)
Comment (simahawk) [reply edi_exchange_template_oca/models/edi_exchange_template_mixin.py:100]: they are similar, not the same. One uses `fields.Date` and the other `fields.Datetime` :wink:

File: edi_exchange_template_oca/models/edi_exchange_template_mixin.py (hunk 2 above)
Comment (simahawk) [reply edi_exchange_template_oca/models/edi_exchange_template_mixin.py:20]: TZ I don't think is needed here. I don't see the point of replacing the min time var. What's the pro?

**Pull request merged**